enhancements/autoscaling/provisioning-request-enablement.md

```diff
@@ -0,0 +1,337 @@
+containing the Cluster Autoscaler API package. The resulting custom resource definition
+manifest will be included in the payload manifests for the operator.
+
+The second implementation detail worth noting is the change in default configuration
+for the Cluster Autoscaler. In order to properly reconcile ProvisioningRequests in
+a cluster, the Cluster Autoscaler must be started with the command line flag
+`--enable-provisioning-requests=true`. The Cluster Autoscaler Operator will be
+updated to include this flag as a default. There will not be a corresponding
+API field to allow user configuration of this flag.
+
+### Risks and Mitigations
+
+This change is generally low risk but there are a few concerns to consider:
+
+1. Unauthorized creation of infrastructure resources. Because the ProvisioningRequest
+  resource will lead to infrastructure resource creation, it must have a proper role and
+  permissions for users to create. This will follow the same role-based authentication
+  control pattern as other Kubernetes resources, but is worth noting as creation of
+  ProvisioningRequests could lead to the creation of infrastructure resources.
+2. Confusion for users around the use of ProvisioningRequests. The Cluster Autoscaler will
+  only react to ProvisioningRequests that are created in the `openshift-machine-api`
+  namespace. This will be clearly illustrated with examples in the OpenShift product
+  documentation.
+
+### Drawbacks
+
+One drawback to this approach is the inclusion of ProvisioningRequest as a supported type
+on core OpenShift. This adds complexity to OpenShift not only in operation but in support
+as well. It is entirely possible that users could install the ProvisioningRequest
+custom resource definition on their own.
+
+The trade-offs for having the user install the ProvisioningRequest definition on their own
+come in two areas: user experience, and Cluster Autoscaler configuration.
+
+The user experience for managing custom resource definitions that interact with Kubernetes
+components, such as the Cluster Autoscaler, requires users to understand the version
+changes and differences. Users need to understand when the project versions have changed
+to ensure that they are installing the correct versions of an API resource. Further, they
+will need to manually migrate on upagrades of the underlying platform.
+
+In addition to installing the ProvisioningRequest, users will need to configure the Cluster
+Autoscaler to reconcile the resources as well. This will require the addition of a new
+field in the ClusterAutoscaler resource on OpenShift.
+
+## Open Questions
+
+1. Do we want to create a field in the ClusterAutoscaler resource to allow disabling of the
+  ProvisioningRequest?
+
+## Test Plan
+
+* Unit tests in openshift/cluster-autoscaler-operator to confirm deployment configuration.
+* End-to-end tests openshift/cluster-api-actuator-pkg to confirm the ProvisioningRequest
```

Comment:
I'd love to sync up with you once this is testable so we confirm that Kueue Operator could use this.

I don't know where to put that in this doc.

Reply:
i'll find a place to note that we will test with kueue before release

Reply:
added a line about kueue integration